specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,232 @@
+# Background
+The WebView2 team has been asked for an API to get the response for a web
+resource as it was received and to provide request headers not available when
+`WebResourceRequested` event is raised (such as HTTP Authentication headers).
+The `WebResourceResponseReceived` event provides such response representation
+and exposes the request as committed.
+
+In this document we describe the new API. We'd appreciate your feedback.
+
+# Description
+The `WebResourceResponseReceived` event allows developers to inspect the
+response object from URL requests (such as HTTP/HTTPS, file and data). A key
+scenario is to allow developers to get Auth headers from an HTTP response to
+authenticate other tools they're using, since the Auth headers are not exposed
+in the `WebResourceRequested` event.
+
+This event is raised when the WebView receives the response for a request for a
+web resource. It provides access to both the response as it was received and the
+request as it was committed, including modifications made by the network stack
+(such as the adding of HTTP Authorization headers). The app can use this event
+to view the actual request and response for a web resource, but modifications
+made to these objects are ignored.
```

Comment:
Are they truly "ignored', or do they simply have no effect on the request?

If they are truly "ignored", then they expected behavior is
```c#
WebView.WebResourceResponseReceived += (s, e) =>
{
   var originalMethod = e.Request.Method;
   e.Request.Method = "Garbage"; // this "set" operation is ignored; method remains unchanged
   Debug.Assert(e.Request.Method == originalMethod);
};

WebView.WebResourceRequested += (s, e) =>
{
  Debug.Assert(e.Request.Method != "Garbage"); // not modified by WebResourceResponseReceived
};
```

I suspect, however, that both assertions will fail. The modification to the Request is retained and is visible to WebResourceRequested. However, it has no effect on the request itself, since the request has already been submitted.

This distinction is important for the Response: If `WebResourceResponseReceived` modifies the `Response` (say, by altering a header), is that modification visible to `WebResourceRequested`?

Reply:
Modifications are set but have no effect on processing the response (nor the request as it was already sent). The response object is created when raising the `WebResourceResponseReceived` event, so `WebResourceRequested` doesn't know about it.